sensor/common/compliance/node_inventory_handler_impl.go

```diff
@@ -79,7 +79,9 @@ func (c *nodeInventoryHandlerImpl) Notify(e common.SensorComponentEvent) {
 	case common.SensorComponentEventCentralReachable:
 		c.centralReady.Signal()
 	case common.SensorComponentEventOfflineMode:
-		// TODO(ROX-17044): handle node_inventory in offline mode
+		// As Compliance enters a retry loop when it is not receiving an ACK,
+		// make sure we send NACK as soon as we enter offline mode
+		c.centralReady.Reset()
```

Comment:
Sorry, I still need to drill here a bit :) 
The comment is now better, however it says that we should sent NACK but we are not doing it.
I propose to either:
- Leave the comment as it is and send the NACK,
- Change the comment to: `As Compliance enters a retry loop when it is not receiving an ACK, there is no need to do anything when entering offline mode`

(I personally prefer the second option 😉)

Reply:
I think, either way the proposed comment has more clarity, so let me change it.
But I do have a question: 
In [L133 ](https://github.com/stackrox/stackrox/blob/mm/ROX-17321-nodescan-nack/sensor/common/compliance/node_inventory_handler_impl.go#L133) of this handler we check for `!c.centralReady.IsDone()` and send a NACK.
By calling `c.centralReady.Reset()`, I would have expected Sensor to now enter the condition in L133 and send a NACK to Compliance. Is my assumption here wrong?

Reply:
Oh yes, that is correct. However, that NACK is sent as a response for an arrival of node-inventory, not for the transition to offline mode. With my comment, I want to say that there is no NACK on transition to offline mode, but you are right, the other NACK is also optional. (the entire NACK message could be removed actually, but we want to wait with that a bit, as the removal itself is trivial).